srml/system/src/lib.rs

```diff
@@ -580,6 +585,12 @@ impl<T: Trait> Module<T> {
 			digest.push(item);
 		}
 
+		// move block hash pruning window by one block
+		let block_hash_count = <BlockHashCount<T>>::get();
+		if number > block_hash_count {
```

Comment:
```suggestion
		if number >+ block_hash_count {
```

Sir, I'm unsure if I'm to tired or not, but I think it should be this way.

Given we have number `101` and `100` as block hash count. This means we already produced `102` blocks.

Reply:
If number is `102` then it means that we have just inserted the hash for its parent `101`. And so we will remove `102 - 100 - 1` i.e. block `1`. So our block hash mapping will have the range `[2, 101]` (100 blocks).

Reply:
I should read the other source code before commenting :D